pkg/controller/elasticsearch/user/predefined_test.go

```diff
@@ -313,7 +313,7 @@ func Test_reconcileInternalUsers(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			c := k8s.NewFakeClient(tt.existingSecrets...)
 			es := tt.es()
-			got, err := reconcileInternalUsers(context.Background(), c, es, tt.existingFileRealm, testPasswordHasher, metadata.Propagate(&es, metadata.Metadata{Labels: es.GetIdentityLabels()}))
+			got, err := reconcileInternalUsers(context.Background(), c, es, tt.existingFileRealm, testPasswordHasher, fixtures.MustTestRandomGenerator(17), metadata.Propagate(&es, metadata.Metadata{Labels: es.GetIdentityLabels()}))
```

Comment:
As mentioned before: why are we testing with different password lengths if the length is not something we assert during the test?

Either:
* use a default length (e.g. by making the argument for `MustTestRandomGenerator` optional) 
* assert the length.

Reply:
I had though this comment was old/invalid with the changes that have been made recently. My apologies. I've added assertions for password length for the tests that do not have pre-existing passwords.